test/empire_widget_test.dart

```diff
@@ -86,147 +87,222 @@ class _MyWidgetState extends EmpireState<_MyWidget, _TestViewModel> {
   }
 }
 
-void main() {
-  late _MyWidget mainWidget;
-  late _TestViewModel viewModel;
-  late _ApplicationViewModel appViewModel;
-  late _ApplicationSubViewModel subViewModel;
-  setUp(() {
-    viewModel = _TestViewModel();
-    appViewModel = _ApplicationViewModel();
-    subViewModel = _ApplicationSubViewModel();
-    mainWidget = _MyWidget(
-      viewModel: viewModel,
-      applicationViewModel: appViewModel,
-      appSubViewModel: subViewModel,
-    );
-  });
-
-  testWidgets(
-      'EmpireWidget Test - Finds Correct Text Widget After Property Change',
-      (tester) async {
-    viewModel.firstName("John");
-    await tester.pumpWidget(mainWidget);
-
-    expect(find.text("John"), findsOneWidget);
-
-    viewModel.firstName("Bob");
-    await tester.pumpAndSettle();
-
-    expect(find.text("Bob"), findsOneWidget);
-  });
-
-  testWidgets('Empire App State Test - Widgets Update on App View Model Change',
-      (tester) async {
-    await tester.pumpWidget(mainWidget);
-
-    final text = find.text("false");
-    expect(text, findsOneWidget);
-
-    appViewModel.change();
-
-    await tester.pumpAndSettle();
-
-    final textTwo = find.text("true");
-    expect(textTwo, findsOneWidget);
-  });
-
-  testWidgets('Update More Than One Property - All Widgets Update',
-      (tester) async {
-    const initialFirstName = 'John';
-    const initialLastName = 'Smith';
-    const initialAge = 88;
-
-    viewModel.firstName(initialFirstName);
-    viewModel.lastName(initialLastName);
-    viewModel.age(initialAge);
-
-    await tester.pumpWidget(mainWidget);
-
-    expect(find.text(initialFirstName), findsOneWidget);
-    expect(find.text(initialLastName), findsOneWidget);
-    expect(find.text(initialAge.toString()), findsOneWidget);
-
-    const newFirstName = 'Bob';
-    const newLastName = 'Brown';
-    const newAge = 20;
-
-    viewModel.setMultiple([
-      {viewModel.firstName: newFirstName},
-      {viewModel.lastName: newLastName},
-      {viewModel.age: newAge},
-    ]);
-
-    await tester.pumpAndSettle();
-
-    expect(find.text(newFirstName), findsOneWidget);
-    expect(find.text(newLastName), findsOneWidget);
-    expect(find.text(newAge.toString()), findsOneWidget);
-  });
+class _TestPreventReassignViewModelWidget extends EmpireWidget<_TestViewModel> {
```

Comment:
Looks like a large change to all the tests, but it's not. I created separate tests groups. One group will run the setUp step, which creates new instances of a test widget and associated view models before each test run. The other group does not as not all the setup objects are required.

The only new UI test is the last test in the new test group.